src/prefect/deployments/runner.py

```diff
@@ -806,10 +806,18 @@ def local_flow():
     )
     # carve out an exception for block based work pools that only have a block in their base job template
     if not is_docker_based_work_pool and not is_block_based_work_pool:
-        raise ValueError(
+        error_message = (
             f"Work pool {work_pool_name!r} does not support custom Docker images. "
             "Please use a work pool with an `image` variable in its base job template."
         )
+        if work_pool.type == "process":
```

Comment:
is there an instance where this wouldn't be process? mostly just curious if we want to test for that

Reply:
surfacing sync discussion, removing the redundant `if work_pool.type == "process":` check here to keep it simple, since the existing case effectively restricts us to that case already

Reply:
added in [eb2c3a5](https://github.com/PrefectHQ/prefect/pull/11615/commits/eb2c3a57c57a62ca50d67290467cb6a0a9332e82)